internal/p2p/peermanager.go

```diff
@@ -832,7 +941,15 @@ func (m *PeerManager) Advertise(peerID types.NodeID, limit uint16) []NodeAddress
 		}
 	}
 
-	return addresses
+	rand.Shuffle(len(addresses), func(i, j int) {
+		addresses[i], addresses[j] = addresses[j], addresses[i]
```

Comment:
Can we come up with a way to make this include peers in the final list with probability according to their score? At the moment, this algorithm sends the number 1 rated peer and the (limit * 2)th peer with the same probability.

Reply:
I think we can, and there are

Reply:
I did something gross but I think workable.

Reply:
I don't think we need this shuffle anymore?

Reply:
removed